### PR TITLE
build: upgrade to go 1.19

### PR DESCRIPTION
**Pull request by fjl**
This changes the CI / release builds to use the latest Go version. It also upgrades golangci-lint to a newer version compatible with Go 1.19.

In Go 1.19, godoc has gained official support for links and lists. Also, the syntax for code blocks in doc comments has changed and now requires a leading tab character. gofmt adapts comments to the new syntax automatically, so there are a lot of comment re-formatting changes in this PR. We need to apply the new format in order to pass the CI lint stage with Go 1.19.

With the linter upgrade, I have decided to disable 'gosec' - it produces too many false-positive warnings. The 'deadcode' and 'varcheck' linters have also been removed because golangci-lint warns about them being unmaintained. 'unused' provides similar coverage and we already have it enabled, so we don't lose much with this change.

### CODE REVIEW

File: accounts/abi/bind/backends/simulated_test.go [hunk 1]
```diff
@@ -1205,10 +1207,11 @@ func TestFork(t *testing.T) {
 Example contract to test event emission:
 
 pragma solidity >=0.7.0 <0.9.0;
-contract Callable {
-    event Called();
-    function Call() public { emit Called(); }
-}
+
```
Comment (holiman) [inline accounts/abi/bind/backends/simulated_test.go:1210]: the `pragma` is part of the code.

File: core/vm/gas_table.go [hunk 2]
```diff
@@ -119,18 +119,18 @@ func gasSStore(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySi
...
+	//   (2.1.) If original value equals current value (this storage slot has not been changed by the current execution context)
+	//     (2.1.1.) If original value is 0, 20000 gas is deducted.
+	//	   (2.1.2.) Otherwise, 5000 gas is deducted. If new value is 0, add 15000 gas to refund counter.
+	//	(2.2.) If original value does not equal current value (this storage slot is dirty), 200 gas is deducted. Apply both of the following clauses.
+	//	  (2.2.1.) If original value is not 0
+	//       (2.2.1.1.) If current value is 0 (also means that new value is not 0), remove 15000 gas from refund counter. We can prove that refund counter will never go below 0.
+	//       (2.2.1.2.) If new value is 0 (also means that current value is not 0), add 15000 gas to refund counter.
+	//	  (2.2.2.) If original value equals new value (this storage slot is reset)
+	//       (2.2.2.1.) If original value is 0, add 19800 gas to refund counter.
+	//		 (2.2.2.2.) Otherwise, add 4800 gas to refund counter.
```
Comment (fjl) [inline core/vm/gas_table.go:133]: Somehow a tab got introduced in this line. Need to fix it.

File: eth/downloader/queue.go [hunk 3]
```diff
@@ -480,9 +480,10 @@ func (q *queue) ReserveReceipts(p *peerConnection, count int) (*fetchRequest, bo
 // to access the queue, so they already need a lock anyway.
 //
 // Returns:
-//   item     - the fetchRequest
-//   progress - whether any progress was made
-//   throttle - if the caller should throttle for a while
+//
+//	item     - the fetchRequest
+//	progress - whether any progress was made
+//	throttle - if the caller should throttle for a while
```
[the remark this answers is not in the document]
Comment (fjl) [reply eth/downloader/queue.go:486]: I will keep these as-is because it looks better with the alignment. It's all internal docs anyway...

**Review by Debu976116 — COMMENTED** [review]
ap

**Comment by Birblewin** [comment]
test 7 passed / 8 & 9 failed - two 2 why? change your IDE?

**Pull request merged**